### PR TITLE
Rewrite code of ModsImporter

### LINKED ISSUE #9537: Rewrite code of ModsImporter
The class `org.jabref.logic.importer.fileformat.ModsImporter` uses JAXB, but can be rewritten using a [StAX-Parser](https://mkyong.com/java/how-to-read-xml-file-in-java-stax-parser/) and thus getting rid of a JAXB dependency.

The current code uses JAXB's internal data to get to the XML element. Example:

```java
        for (JAXBElement<?> jaxbElement : topicOrGeographicOrTemporal) {
            Object value = jaxbElement.getValue();
            String elementName = jaxbElement.getName().getLocalPart();
```

This can be directly done using StAX' capabilities.

### PR DESCRIPTION
**Pull request by aqurilla**
This fixes #9537 by rewriting the `org.jabref.logic.importer.fileformat.ModsImporter` class to use a StAX-Parser, and removes all JAXB dependencies from the class

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/logic/importer/fileformat/mods/IdentifierDefn.java
Comment (calixtus) [inline src/main/java/org/jabref/logic/importer/fileformat/mods/IdentifierDefn.java:29]: If you inject the values in the constructor, this and the other `Defn` could be records.
Also please avoid Abbreviations in the classname.

**Comment by calixtus** [comment]
Thanks for taking care of this one. Would be great to have some unnecessary dependencies cleaned up.
Looking forward seeing this completed!

**Comment by aqurilla** [comment]
Thanks for the feedback @calixtus, I have made the requested changes in the latest commit

File: src/main/java/org/jabref/logic/importer/fileformat/ModsImporter.java
Comment (calixtus) [inline src/main/java/org/jabref/logic/importer/fileformat/ModsImporter.java:535]: Instead of building a custom year parser, you could use `our model.entry.Date.parse()` method. That way, we can avoid unnecessary data loss by parsing.

Not sure about created and modified field. We use the fields `CREATIONDATE` and `MODIFICATIONDATE` for internal description of changes to bibentries. Question is: Should we import them from the source and interpret them like our internal ones, or should we forget about them @koppor @Siedlerchr ?

Comment (aqurilla) [reply src/main/java/org/jabref/logic/importer/fileformat/ModsImporter.java]: Updated to use the `model.entry.Date.parse()` method

Comment (calixtus) [reply src/main/java/org/jabref/logic/importer/fileformat/ModsImporter.java]: Can you fill the date field in biblatex mode too? In biblatex mode, this is the preferred way to keep dates (you could use getNormalized). Also the month field could could be filled, if there is a month available...

**Review by calixtus — APPROVED** [review]
Thanks for your changes. Besides minor comments, LGTM :+1:. The changes need a second review and thus this PR awaits a second review.

File: src/main/java/org/jabref/logic/importer/fileformat/ModsImporter.java
Comment (aqurilla) [reply src/main/java/org/jabref/logic/importer/fileformat/ModsImporter.java]: Thanks, I missed that - I'll add these fields soon

Comment (aqurilla) [reply src/main/java/org/jabref/logic/importer/fileformat/ModsImporter.java]: Hi @calixtus, I've added the date and month fields. I've also modified a couple of test files since they do not include the `date` field currently

**Review by calixtus — APPROVED** [review]

**Review by Siedlerchr — APPROVED** [review]

**Comment by Siedlerchr** [comment]
Thanks! LGTM!

File: src/main/java/org/jabref/logic/importer/fileformat/ModsImporter.java
Comment (koppor) [reply src/main/java/org/jabref/logic/importer/fileformat/ModsImporter.java]: > Not sure about created and modified field. We use the fields `CREATIONDATE` and `MODIFICATIONDATE` for internal description of changes to bibentries. Question is: Should we import them from the source and interpret them like our internal ones, or should we forget about them @koppor @Siedlerchr ?

I would opt for mapping `dateCaptured` to `CREATIONDATE` and `dateModified` to `MODIFICATIONDATE`. This now has to go into a follow-up PR.

**Pull request merged**
